File: Java_Generated/Java/untitled/src/measurements/arrows/DataLink.java
package measurements.arrows;

import category_theory.CategoryArrow;
import category_theory.interfaces.ICategoryObject;
import category_theory.interfaces.IDesktop;
import measurements.Measurements;
import measurements.interfaces.IDataConsumer;
import measurements.interfaces.IMeasurements;

public class DataLink extends CategoryArrow {
    public DataLink(String name, IDesktop desktop) {
        super(name, desktop);
    }

    IMeasurements measurements;

    IDataConsumer dataConsumer;

    @Override
    public void setSource(ICategoryObject source) {
        super.setSource(source);
        dataConsumer = (IDataConsumer) source;
    }

    /**
     * @param target
     */
    @Override
    public void setTarget(ICategoryObject target) {
        super.setTarget(target);
        measurements = (Measurements) target;
        dataConsumer.addMeasurements(measurements);
    }

}

File: src/Python/measurements/datalink.py
from ..category_object import CategoryObject
from ..category_arrow import CategoryArrow
from ..desktop import Desktop
from ..measurements.measurements import Measurements
from ..measurements.measurement import Measurement

class DataLink(CategoryArrow):

    def __init__(self, name, desktop):
        super().__init__(name, desktop)
    
    @property
    def name(self):
        return self._name
    
    @property
    def source(self):
        return self._source
    
    @source.setter
    def source(self, source: CategoryObject):
        self._source = source

    @property
    def target(self):
        return self._source
    
    @target.setter
    def source(self, target: CategoryObject):
        self._target = target
